cli/src/config.rs

```diff
@@ -259,11 +259,31 @@ pub struct ConfigEnv {
 impl ConfigEnv {
     /// Initializes configuration loader based on environment variables.
     pub fn from_environment() -> Self {
+        #[cfg(not(target_os = "macos"))]
+        let config_dir = etcetera::choose_base_strategy().ok().map(|s| {
+            use etcetera::BaseStrategy as _;
+            s.config_dir()
+        });
+        // older versions of jj used the more "GUI" config option due to dirs,
+        // which is not really correct for CLI applications
+        #[cfg(target_os = "macos")]
+        let config_dir = etcetera::base_strategy::choose_native_strategy()
+            .ok()
+            .map(|s| {
+                use etcetera::BaseStrategy as _;
+                // note that etcetera calls Library/Application Support the "data dir",
+                // Library/Preferences is supposed to be exclusively plists
+                s.data_dir()
+            });
+
```

Comment:
Nit: add a warning that this behavior is deprecated and will be removed in 6 releases and a TODO for the code removal.